tox.ini

```diff
@@ -294,23 +292,41 @@ deps =
     {[testenv:build]deps}
     twine
 passenv =
+    # Intentionally omit TWINE_REPOSITORY_URL from the passenv list,
+    # as this overrides other ways of setting the repository and could
+    # unexpectedly result in releasing to the wrong repo
     {[testenv]passenv} \
     TWINE_USERNAME \
-    TWINE_PASSWORD \
-    TWINE_REPOSITORY_URL
+    TWINE_PASSWORD
 commands =
     {[testenv:build]commands}
-    twine upload --skip-existing --repository testpypi dist/*
+
+[testenv:release-private]
+basepython = python3
+skip_install = true
+deps = {[testenv:release-base]deps}
+passenv =
+    {[testenv:release-base]passenv} \
+    TWINE_REPOSITORY_URL
+setenv =
+    # Explicitly set the URL as the env variable value, which will cause us to
+    # throw an error if the variable is not set. Otherwise, omission of the
+    # env variable could cause us to unintentionally upload to the wrong repo
+    TWINE_REPOSITORY_URL = {env:TWINE_REPOSITORY_URL}
+commands =
+    {[testenv:release-base]commands}
+    # Omitting an explicit repository will cause twine to use the repository
+    # specified in the environment variable
```

Comment:
Need the `twine` command here as well since we're not setting it in the base target anymore.